test/jdk/javax/swing/JTable/8236907/LastVisibleRow.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/*
+ * @test
+ * @key headful
+ * @bug 8236907
+ * @summary  Verifies if Jtable last row is visible.
+ * @run main LastVisibleRow
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Container;
+import java.awt.Dimension;
+import java.awt.event.InputEvent;
+import java.awt.image.BufferedImage;
+import java.text.MessageFormat;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+
+import javax.print.attribute.HashPrintRequestAttributeSet;
+import javax.print.attribute.PrintRequestAttributeSet;
+
+import javax.swing.BorderFactory;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JTable;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+
+public class LastVisibleRow {
+    static JFrame frame;
+    static JDialog dialog;
+    static JTable table;
+    static Robot testRobot;
+    private static BufferedImage bufferedImageAfter;
+    private static BufferedImage bufferedImageBefore;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            testRobot = new Robot();
+
+            javax.swing.SwingUtilities.invokeAndWait(new Runnable() {
+
+                public void run() {
+                    try {
+                        createAndShowGUI();
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                }
+            });
+            testRobot.waitForIdle();
+            CaptureBeforeClick();
+            testRobot.waitForIdle();
+            MouseActions();
+            testRobot.waitForIdle();
+            clearSelect();
+            testRobot.waitForIdle();
+            CaptureAfterClick();
+
+            if (!compare(bufferedImageBefore, bufferedImageAfter)) {
```

Comment:
You probably need to wait a little using robot.delay() after you capture the bufferedimage and then do the comparison.
I hope you have ran this test on CI on all platforms.

Reply:
Yeah, I've run CI on all platforms, all have passed.
https://mach5.us.oracle.com/mdash/jobs/tejer-jdk-20220217-0621-29214728?search=result.status%3APASSED